### PR TITLE
Add `plot_expected_purchases` function

### PR DESCRIPTION
**Pull request by ColtAllen**
## Description

This PR consolidates the `plot_cumulative_transactions` and `plot_incremental_transactions` functions from `lifetimes` into a single plotting function, which is very useful for model evaluation:

image

So useful in fact that when I added it to the Quickstart, it revealed a data discrepancy causing model bias! I left it in the notebook because I felt like it was a teachable moment, but at the same time we're showcasing this library with a skewed model. I'm curious to hear your thoughts on this.

I added `plt.clf()` to the plotting function to reset the `pyplot` object between plots, which is useful when logging multiple figures to MLflow. I also took the opportunity to make a few other revisions to the Quickstart, such as use of the `Prior` class in model configs and a watermark. 

## Related Issue

- [ ] Closes #
- [x] Related to  https://github.com/pymc-labs/pymc-marketing/issues/326

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [x] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [ ] MMM
- [x] CLV

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [x] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--1115.org.readthedocs.build/en/1115/

### CODE REVIEW

File: pymc_marketing/clv/plotting.py [hunk 1]
```diff
@@ -344,15 +346,138 @@ def plot_probability_alive_matrix(
...
+    time_unit: str = "D",
+    time_scaler: float | None = 1,
+    sort_transactions: bool | None = True,
+    set_index_date: bool | None = False,
+    title: str | None = None,
+    xlabel: str = "Time Periods",
+    ylabel: str = "Purchases",
+    ax: plt.Axes | None = None,
+    **kwargs,
+):
```
Comment (juanitorduz) [inline pymc_marketing/clv/plotting.py:375]: Can we add the return type hint `plt.Axes` (or shall we return the Figure object instead?, we do this for the MMM plots)

File: pymc_marketing/clv/plotting.py [hunk 2]
```diff
@@ -344,15 +346,138 @@ def plot_probability_alive_matrix(
...
+        purchase_history,
+        customer_id_col,
+        datetime_col,
+        t,
+        datetime_format,
+        time_unit,
+        time_scaler,
+        sort_transactions,
+        set_index_date,
+    )
```
Comment (juanitorduz) [inline pymc_marketing/clv/plotting.py:457]: This function has a lot of arguments. Shall we use keyword arguments to make sure we do not pass the wrong order

```python
    df_cum_transactions = _expected_cumulative_transactions(
        model=model,
        ...
    )
```

**Review by juanitorduz — APPROVED** [review]
I really like this type of diagnostic plots!

I left some suggestions :D

File: pymc_marketing/clv/plotting.py (hunk 2 above)
Comment (wd60622) [reply pymc_marketing/clv/plotting.py:457]: +1

File: pymc_marketing/clv/plotting.py [hunk 3]
```diff
@@ -344,15 +346,138 @@ def plot_probability_alive_matrix(
...
+    datetime_col :  string
+        Column in the *transactions* DataFrame denoting datetimes purchase were made.
+    t : int
+        Number of time units since earliest purchase to include in plot.
+    plot_cumulative : bool
+        Default: *True*
+        Plot cumulative purchases over time. Set to *False* to plot incremental purchases.
+    t_cal : int, optional
+        If testing model on unobserved data, specify number of time units in training data to add an indicator for
+        the start of the testing period.
```
Comment (wd60622) [inline pymc_marketing/clv/plotting.py:403]: I am having a bit of a hard time understand the name `t_cal`. Could be the suffix `cal` throwing me off.

**Review by wd60622 — COMMENTED** [review]
Overall looks good. Just some comments on the verbiage

File: pymc_marketing/clv/plotting.py (hunk 3 above)
Comment (ColtAllen) [reply pymc_marketing/clv/plotting.py:403]: Ah, a `lifetimes` leftover I overlooked. This marks the start of unobserved data if evaluating the model with `rfm_train_test_split`. I've renamed it to `t_unobserved`.

File: pymc_marketing/clv/plotting.py [hunk 4]
```diff
@@ -344,15 +346,138 @@ def plot_probability_alive_matrix(
...
+    axes: matplotlib.AxesSubplot
+
+    References
+    ----------
+    .. [1] Fader, Peter S., Bruce G.S. Hardie, and Ka Lok Lee (2005),
+    A Note on Implementing the Pareto/NBD Model in MATLAB.
+    http://brucehardie.com/notes/008/
+    """
+    # clear any existing pyplot figures
+    plt.clf()
```
Comment (wd60622) [inline pymc_marketing/clv/plotting.py:441]: I understand where you are coming from here but shouldn't it be on the user to do this? Also, does this obstruct the passing of the ax?
```suggestion
```

Comment (ColtAllen) [reply pymc_marketing/clv/plotting.py]: You may have a point about the `ax` arg, but the unit tests will error out if this isn't provided, so I've moved it out of the function and into the test.

**Comment by juanitorduz** [comment]
Thanks @ColtAllen !

**Pull request merged**